### ISSUE TITLE
Importing google_compute_project_metadata_item fails with "Error: project: required field is not set"

### ISSUE THREAD

**Issue by pdecat** [terraform-providers/terraform-provider-google#5742]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

```
# terraform version
Terraform v0.12.21
+ provider.google v3.9.0
```

Note: same issue happens with provider version 2.20.2.

### Affected Resource(s)

* `google_compute_project_metadata_item`

### Terraform Configuration Files

```tf
provider "google" {
  version = "3.9.0"

  project = "myproject"
  region  = "europe-west1"
}

resource "google_project" "project" {
  name            = "myproject"
  project_id      = "myproject"
  billing_account = "123456-123456-123456"
}

resource "google_compute_project_metadata_item" "project_disable_legacy_endpoints" {
  project = google_project.project.project_id

  key   = "disable-legacy-endpoints"
  value = "TRUE"
}
```

### Debug Output

https://gist.github.com/pdecat/95b65fcc68f834bc42837a3d79d62b7c

### Panic Output

N/A

### Expected Behavior

Import should be successful.

### Actual Behavior

Import fails:
```sh
# terraform import google_compute_project_metadata_item.project_disable_legacy_endpoints disable-legacy-endpoints
google_compute_project_metadata_item.project_disable_legacy_endpoints: Importing from ID "disable-legacy-endpoints"...
google_compute_project_metadata_item.project_disable_legacy_endpoints: Import prepared!
  Prepared google_compute_project_metadata_item for import
google_compute_project_metadata_item.project_disable_legacy_endpoints: Refreshing state... [id=disable-legacy-endpoints]

Error: project: required field is not set
```

### Steps to Reproduce

1. `terraform import google_compute_project_metadata_item.project_disable_legacy_endpoints disable-legacy-endpoints`

### References

* #4610
* #5550

**Comment by pdecat** [comment]
Hmm, this issue does not happen with an new empty state:

```
# terraform import google_compute_project_metadata_item.project_disable_legacy_endpoints disable-legacy-endpoints
google_compute_project_metadata_item.project_disable_legacy_endpoints: Importing from ID "disable-legacy-endpoints"...
google_compute_project_metadata_item.project_disable_legacy_endpoints: Import prepared!
  Prepared google_compute_project_metadata_item for import
google_compute_project_metadata_item.project_disable_legacy_endpoints: Refreshing state... [id=disable-legacy-endpoints]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.
```

Mine was just upgraded from 2.20.2 to 3.9.0.

**Comment by pdecat** [comment]
Import also works with an empty state when resources are in a module:

`main.tf`:
```tf
provider "google" {
  version = "3.9.0"

  project = "myproject"
  region  = "europe-west1"
}

module "root_state" {
  source = "./root_state"
}
``` 

`root_state/main.tf`:
```tf
resource "google_project" "project" {
  name            = "myproject"
  project_id      = "myproject"
  billing_account = "123456-123456-123456"
}

resource "google_compute_project_metadata_item" "project_disable_legacy_endpoints" {
  project = google_project.project.project_id

  key   = "disable-legacy-endpoints"
  value = "TRUE"
}
```

```sh
# terraform import module.root_state.google_compute_project_metadata_item.project_disable_legacy_endpoints disable-legacy-endpoints
module.root_state.google_compute_project_metadata_item.project_disable_legacy_endpoints: Importing from ID "disable-legacy-endpoints"...
module.root_state.google_compute_project_metadata_item.project_disable_legacy_endpoints: Import prepared!
  Prepared google_compute_project_metadata_item for import
module.root_state.google_compute_project_metadata_item.project_disable_legacy_endpoints: Refreshing state... [id=disable-legacy-endpoints]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.
```

**Comment by edwardmedia** [comment]
@pdecat you provided two cases in which import works for `empty state`. That is expected. If I leave the state file in place, I got below error, which is not the same as you got. What is the scenario for NON `empty state`? Can you specify the steps how you hit your error and what exact config do you have then? 
```
google_compute_project_metadata_item.project_disable_legacy_endpoints: Importing from ID "my_metadata"...
google_compute_project_metadata_item.project_disable_legacy_endpoints: Import prepared!
  Prepared google_compute_project_metadata_item for import

Error: Resource already managed by Terraform

Terraform is already managing a remote object for
google_compute_project_metadata_item.project_disable_legacy_endpoints. To
import to this address you must first remove the existing object from the
state.
```

**Comment by pdecat** [comment]
I just added that `google_compute_project_metadata_item` part this morning to catch up with out of band changes I noticed:

```
resource "google_compute_project_metadata_item" "project_disable_legacy_endpoints" {
  project = google_project.project.project_id

  key   = "disable-legacy-endpoints"
  value = "TRUE"
}
```

The first version of the state for that stack was created in 2016 (terraform 0.7.4 at the time, before the big provider split) and was upgraded to almost all versions of terraform and the google provider since then.

I wanted to import the project metadata item resource to avoid updating the Google Compute Metadata and faced the issue described here.

**Comment by edwardmedia** [comment]
@pdecat  can you try to import if you have the config like below, then manually add the key / value in the config accordingly?
```
resource "google_compute_project_metadata_item" "project_disable_legacy_endpoints" {
 project = google_project.project.project_id
}
```

**Comment by pdecat** [comment]
@edwardmedia, thanks for the help on this issue!

I've tried commenting these fields but the same issue still happened.

Doing further tests, I've managed to locate the cause.
I had this in my `root_state` module that came from pre-0.11 configuration (we fixed this in our other modules a long time ago):

```
provider google {
  version = "~> 3.9.0"
}
```

The underlying issue is described here: https://github.com/hashicorp/terraform/issues/16835

I replaced it with the following and the import worked:

```
terraform {
  required_providers {
    google = "~> 3.9.0"
  }
}
```

I guess we can close this issue.

Best regards,
Patrick.
